tensorflow_io/avro/20190307-avro-dataset.md

```diff
@@ -0,0 +1,299 @@
+# Avro Dataset for TensorFlow IO
+
+| Status        | Accepted                                                          |
+| :------------ | :---------------------------------------------------------------- |
+| **Author(s)** | florian.raudies@gmail.com                                         |
+| **Sponsor**   | N/A                                                               |
+| **Updated**   | 2019-03-07                                                        |
+| **Obsoletes** | N/A                                                               |
+
+## Overview
+
+Load [Avro](https://avro.apache.org/docs/current/) formatted data natively into  [TensorFlow](https://www.tensorflow.org/) from file systems that are supported by TensorFlow, e.g. the Hadoop Distributed File System ([HDFS](https://hadoop.apache.org/docs/r1.2.1/hdfs_design.html)).
+
+## Motivation
+
+Loading Avro formatted data will enable the data science community to leverage TensorFlow more easily. Currently, loading Avro formatted data into TensorFlow is not supported. Thus, ways of getting data into TensorFlow requires a conversion to other data formats, e.g. protobuf, CSV, or proprietary formats. Such custom solutions are difficult to maintain and do not generalize well for other uses.
+
+## Design Proposal
+
+The Avro dataset shall support
+
+- Transform primitive and nested Avro types into TensorFlow sparse and dense tensors
+- The creation of dimensions by (see also examples in Tables 2 and 3)
+  - Combining of several rows -- also called batching
+  - Combining several Avro fields to define indices and values for sparse tensors
+  - Combining and reshaping of elements of an Avro non-nested array into a multi-dimensional tensor using row-major-order
+  - Nestings of Avro typed arrays
+- Filling in scalars or tensors for missing values whenever the number of provided elements does not match the desired number of elements by using default values -- if no defaults or the wrong defaults are provided this results in a runtime error surfaced through an exception in python and erroneous status returned in c/c++
+- Full support for N-dimensional dense tensors and N-dimensional sparse tensors
+- A mapping of Avro primitive types to TensorFlow primitive types (see Table 1)
+- For Avro&#39;s null type the default shall be assumed if provided; otherwise fail
```

Comment:
Does regular `'` work instead of `&#39;`? I see a `'` at the end of the document.